RIOT-OS/RIOT
# examples/gnrc_border_router: move module specific config to extra Makefile

### Contribution description
This cleans up the `gnrc_border_router` examples Makefile, to be more usable for new users again. This mostly moves configuration to a `Makefile.<feature>.conf`, so it is out of sight in the main Makefile. The only two things I changed are
1. use the capability for `start_network.sh` to not use UHCP from #13482 
2. uncomment the SLIP part and make it configurable at call-time of `make` using `USE_SLIP`



### Testing procedure
Compile and try all permutations of the following `make` parameters

- `BOARD=native` (or other)
- `USE_DHCPV6=1` (or 0)
- `USE_SLIP=1` (or 0)

All should compile and work (for `USE_SLIP=1` a `slipdev_params.h` is required, you can take that from the `gcoap` example e.g.; use `tunslip` with a FTDI adapter to open a SLIP connection).



### Issues/PRs references
Requires #13482 
See https://github.com/RIOT-OS/RIOT/pull/13424#issuecomment-590792281
